### ISSUE TITLE
Potential Gating Network Coding Error (on sigmoid) in armo-rm Stage 2

### ISSUE THREAD

**Issue by SWY666** [RLHFlow/RLHF-Reward-Modeling#54]
Thanks for sharing the great code!

In your armo-rm stage 2 code line 80, you add sigmoid on the output at the dim 1, which is shown as follows:

```
    def forward(self, x: torch.FloatTensor) -> torch.FloatTensor:
        # Apply the linear layers with ReLU and dropout
        for i, layer in enumerate(self.layers):
            x = layer(x)
            if i < len(self.layers) - 1:
                x = F.relu(x)
                if self.dropout_prob > 0:
                    x = F.dropout(x, p=self.dropout_prob, training=self.training)
        # Apply softmax with temperature scaling
        x = F.softmax(x / self.temperature, dim=1) # This line
        return x * self.logit_scale[0]
```

But in practice the shape of the x here has 3 dimensions (i.e., batchsize, 2, 19; the 2 here is chosen & rejected), so I wonder whether the **dim=1** here should be changed to **dim=-1** to make the softmax be added on the last 19 weights during the training? 

![Image](https://github.com/user-attachments/assets/a57a78d5-516f-47f9-a691-74cea947506a)

Btw, do you use this code for Amom-rm for crafting [RLHFlow/ArmoRM-Llama3-8B-v0.1](https://huggingface.co/RLHFlow/ArmoRM-Llama3-8B-v0.1)? When I use your code (i.e., with softmax set as **dim=1**), almost all 19 weights look weird (they sum not equal to 1, and they are almost the same value), and I cannot reproduce the high reward score as shown in Reward Bench Leaderboard.

**Comment by WeiXiongUST** [comment]
@Haoxiang-Wang hi haoxiang, could you look into this?
